### ISSUE TITLE
Bad reward function on LunarLander

### ISSUE THREAD

**Issue by SamuelePolimi** [openai/gym#473]
The robot receives a reward >100. in the position shown:
![robotup](https://cloud.githubusercontent.com/assets/17749114/22381319/6c0d9256-e4c1-11e6-9305-5eab38c9481a.png)

These are the last sequences of <state, action, reward, done,  info>:

('state: ', array([ 0.66872807,  0.31876364,  0.93155613, -1.29488436, -0.51435429,
        0.1669438 ,  0.        ,  0.        ]))
('action: ', 3)
('reward: ', -2.1935851844794727)
('done: ', False)
('info: ', {})
('state: ', array([ 0.67817574,  0.29887105,  0.93890667, -1.32398567, -0.50774068,
        0.13227191,  0.        ,  0.        ]))
('action: ', 2)
('reward: ', -2.1533557847858331)
('done: ', False)
('info: ', {})
('state: ', array([ 0.68794422,  0.27886062,  0.97039976, -1.33165029, -0.50046468,
        0.14552021,  0.        ,  0.        ]))
('action: ', 3)
('reward: ', -2.8515008774707256)
('done: ', False)
('info: ', {})
('state: ', array([ 0.69780025,  0.2584075 ,  0.98143816, -1.36203613, -0.4958019 ,
        0.09325559,  0.        ,  0.        ]))
('action: ', 1)
('reward: ', -0.98382723840089059)
('done: ', False)
('info: ', {})
('state: ', array([ 0.70758858,  0.23759413,  0.97278767, -1.38539073, -0.48901996,
        0.13563849,  0.        ,  0.        ]))
('action: ', 2)
('reward: ', -2.1572379230373029)
('done: ', False)
('info: ', {})
('state: ', array([ 0.71770306,  0.21670242,  1.00481367, -1.39043999, -0.48160586,
        0.14828198,  0.        ,  0.        ]))
('action: ', 3)
('reward: ', -2.5708908531227892)
('done: ', False)
('info: ', {})
('state: ', array([ 0.72787399,  0.19539009,  1.01184149, -1.41900749, -0.47577116,
        0.11669427,  0.        ,  0.        ]))
('action: ', 2)
('reward: ', 8.7827422640659272)
('done: ', False)
('info: ', {})
('state: ', array([ 0.73852997,  0.17441626,  1.05992413, -1.39633013, -0.46950805,
        0.12526217,  1.        ,  0.        ]))
('action: ', 2)
('reward: ', 124.30866382855071)
('done: ', False)
('info: ', {})

Also, there is the issue that when the robot is in the right position (so standing between the flags, landing softly, and remaining in that position), it doesn't receive the correct reward and neither the episode finishes.

**Comment by olegklimov** [comment]
This is a correct reward. It is received because hitting a mountain with the leg reduces speed.

You just hit one of things that makes this environment interesting. I've observed agents that try to exploit this bad minima at first, by falling down with great speed, one of legs forward. Good agent eventually gets out of it.

When lander is in the right position and shuts down engines, episode reliably ends. This is tested a lot. Episode ends sometimes even with engines still working, good agent learns how to power down engines from probability of that happening.

I can also assure you there are algorithms that solve this problem, that is one of the most simple tasks in Gym. Working solution should take less than 1M, very good solutions 2M frames.

**Comment by SamuelePolimi** [comment]
I trust you, but honestly, it doesn't seem to me a good idea to give such an high reward in that position. I hope that in NASA they don't consider it a good landing too. 
While a lot of time my lander lands in a good position, and it doesn't receive any reward higher than 100. 
In my own opinion this is a bit wired. 
I am using this environment from a while, and I saw it working properly. Then probably I updated something, and now it is no working anymore.
